explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusScheduler.java

```diff
@@ -43,6 +43,13 @@ public Map<UUID, BaseMetricRequest> getSpdRequests() {
         return spdRequests;
     }
 
+    public Map<UUID, BaseMetricRequest> getAllRequests() {
```

Comment:
I'm not sure if we should use a `ConcurrentHashmap` for this. cc @danielezonca @tteofili ?

Reply:
Good point, in general I think we should make these [two fields](https://github.com/trustyai-explainability/trustyai-explainability/blob/6f526778a09041bad8ea19070acfd399e6044988/explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusScheduler.java#L26-L27) `ConcurrentHashMap`